tests/callbacks/Helpers.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef KOKKOS_UTILS_TESTS_HELPERS_HPP
+#define KOKKOS_UTILS_TESTS_HELPERS_HPP
+
+#include "gmock/gmock.h"
+
+#include "kokkos-utils/callbacks/Manager.hpp"
+
+namespace Kokkos::utils::tests::callbacks
+{
+
+class ManagerTestFixture : public virtual ::testing::Test
+{
+public:
+    ManagerTestFixture()  { Kokkos::utils::callbacks::Manager::initialize(); }
+    ~ManagerTestFixture() { Kokkos::utils::callbacks::Manager::finalize(); }
+};
+
+#define DEFINE_EVENT_WITH_NAME_MATCHER(__eventtype__)                                                                                                 \
+    MATCHER_P(__eventtype__##WithName, matcher, "is " #__eventtype__ " with name that " + ::testing::DescribeMatcher<std::string>(matcher, negation)) \
+    {                                                                                                                                                 \
+        using Kokkos::utils::callbacks::__eventtype__;                                                                                                \
+        if (const __eventtype__* parg = std::get_if<__eventtype__>(&arg))                                                                             \
+            return ExplainMatchResult(matcher, parg->name, result_listener);                                                                          \
+        else                                                                                                                                          \
+            return false;                                                                                                                             \
+    }
+
+DEFINE_EVENT_WITH_NAME_MATCHER(BeginParallelForEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(BeginParallelReduceEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(BeginParallelScanEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(BeginFenceEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(CreateProfileSectionEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(PushRegionEvent)
+DEFINE_EVENT_WITH_NAME_MATCHER(ProfileEvent)
+
+namespace impl
+{
+
+template <typename IteratorFirst, typename IteratorLast, typename... Matchers>
+bool contains_in_order(const IteratorFirst& iterator_first, const IteratorLast& iterator_last, const std::tuple<Matchers...>& matchers)
+{
+    return std::apply(
+        [&iterator_first, &iterator_last] <typename Arg, typename... Args> (const Arg& matcher_head, const Args&... matchers_tail) -> bool
+        {
```

Comment:
I'm not sure to understand why you want it as a tuple, and not as a variadic.

Reply:
It was because the `MATCHER_P` macro takes a single argument. And so we can't pass the variadic. But I've changed the implementation. So the issue is outdated.